src/NuGet.Core/NuGet.Packaging/Signing/Archive/CentralDirectoryMetadata.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// This class is used to hold metatdata about the central directory archive structure
+    /// </summary>
+    public class CentralDirectoryMetadata
+    {
+        /// <summary>
+        /// Position in bytes of the corresponding central directory header relative to the start of the archive
+        /// </summary>
+        public long Position { get; set; }
```

Comment:
Should all these types be unsigned?  For fields we look at the ZIP spec only uses unsigned integers.

Reply:
To be able to be CLS-compliant we cannot have public attributes that are unsigned. Also, this positions hold the data gotten from the `reader.BaseStream.Position`, which is also a long.